From 9e0b7f4013ca233dc071c760a84bee06ed61c750 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maxime=20M=C3=A9r=C3=A9?= <maxime.mere@foss.st.com>
Date: Thu, 27 Mar 2025 10:56:38 +0100
Subject: [PATCH 809/841] feat(st): use BL33 parameter x0 for physical address
 of device tree blob
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Use BL33 parameter x0 for physical address of device tree blob (dtb) in
system RAM as expected by Linux kernel.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 6b05d6c80efff375f0ae29ebad34c53b3364f037 feat(st): use BL33 parameter x0 for physical address of device tree blob]
Change-Id: Ic8c997ea59ed299594cb4eb27db546b4e68a783a
Signed-off-by: Maxime Méré <maxime.mere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/447024
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 plat/st/stm32mp2/bl31_plat_setup.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/plat/st/stm32mp2/bl31_plat_setup.c b/plat/st/stm32mp2/bl31_plat_setup.c
index abe636fa2..e1b51dac4 100644
--- a/plat/st/stm32mp2/bl31_plat_setup.c
+++ b/plat/st/stm32mp2/bl31_plat_setup.c
@@ -103,9 +103,9 @@ void bl31_early_platform_setup2(u_register_t arg0, u_register_t arg1,
 			 *  share it to BL33
 			 */
 			if (arg2 != 0U) {
-				bl33_image_ep_info.args.arg0 = 0U;
+				bl33_image_ep_info.args.arg0 = arg2;
 				bl33_image_ep_info.args.arg1 = 0U;
-				bl33_image_ep_info.args.arg2 = arg2;
+				bl33_image_ep_info.args.arg2 = 0U;
 			}
 		}
 
-- 
2.47.2

